### PR TITLE
Clean up event handlers and other subscriptions

### PR DESCRIPTION
**Pull request by davidfowl**
- Also removed the closure when logging
- Captured _applicationStartedRegistration though it shouldn't strictly be necessary since the CTS *should* be disposed when the host is disposed

This change was inspired by looking at memory dumps and seeing this:

image

Maybe worth doing in the WebHost as well (but we're moving away from that)

### CODE REVIEW

File: src/Hosting/Hosting/src/Internal/ConsoleLifetime.cs [hunk 1]
```diff
@@ -72,6 +83,11 @@ public Task StopAsync(CancellationToken cancellationToken)
         public void Dispose()
         {
             _shutdownBlock.Set();
+
+            AppDomain.CurrentDomain.ProcessExit += OnProcessExit;
```
Comment (JunTaoLuo) [inline src/Hosting/Hosting/src/Internal/ConsoleLifetime.cs]: Why are these added again?

Comment (davidfowl) [reply src/Hosting/Hosting/src/Internal/ConsoleLifetime.cs]: Wow fail, should be -, copy pasta

Comment (JunTaoLuo) [reply src/Hosting/Hosting/src/Internal/ConsoleLifetime.cs]: That would make a bit more sense.

Comment (davidfowl) [reply src/Hosting/Hosting/src/Internal/ConsoleLifetime.cs]: 🆙 📅

**Comment by JunTaoLuo** [comment]
No issue filed?

**Comment by davidfowl** [comment]
@JunTaoLuo I remember one from a while back but I can't find it. Funnily enough, I'm trying to reproduce the problem and found another more severe leak..

**Comment by davidfowl** [comment]
After:

image

**Comment by davidfowl** [comment]
Turns out it was holding onto more state that I originally recognized.
